### PR TITLE
fix: enforce token validation if any token keys are defined

### PR DESCRIPTION
**Pull request by fiam**
- feat: add testapp for JWKS
- chore: update MicahParks/keyfunc to v1.9.0

Closes #294

### CODE REVIEW

**Comment by fiam** [comment]
@MicahParks Thaks for your contribution! I've pulled your commits from #294 and I've also added some tests for JWKS on top.

File: testapps/jwks/test/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		});
+		const result = await client.query({
+			operationName: 'Hello',
+		});
+		expectUnauthorized(result);
+	});
+
+	test('empty token', async () => {
+		const client = wg.client();
+		client.setExtraHeaders({
```
Comment (jensneuse) [inline testapps/jwks/test/index.test.ts:91]: Can we add a method on the client e.g. named `client.setAuthorizationToken`.
Extra headers is quite low level. I think setAuthorizationToken give a user better ergonomics.

File: testapps/jwks/test/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+const wg = createTestServer({ fetch: fetch as any });
+
+let authorizationToken = '';
+
+const startServer = async () => {
+	const keyID = '123456';
+	const keyAlgorithm = 'RS256';
+	const { publicKey, privateKey } = await jose.generateKeyPair(keyAlgorithm);
+	const publicKeyJWK = await jose.exportJWK(publicKey);
+	const token = await new jose.SignJWT({})
```
Comment (jensneuse) [inline testapps/jwks/test/index.test.ts:17]: great way to test!

File: testapps/jwks/test/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+				kid: keyID,
+				alg: keyAlgorithm,
+				kty: 'RSA',
+				n: publicKeyJWK.n!,
+				e: publicKeyJWK.e!,
+			},
+		],
+	};
+
+	process.env['JWKS_JSON'] = JSON.stringify(jwksKeys);
```
Comment (jensneuse) [inline testapps/jwks/test/index.test.ts:43]: so smart!

**Review by jensneuse — APPROVED** [review]

**Review by Pagebakers — APPROVED** [review]

File: testapps/jwks/test/index.test.ts (hunk 1 above)
Comment (fiam) [reply testapps/jwks/test/index.test.ts:91]: Sure

File: testapps/jwks/test/index.test.ts (hunk 3 above)
Comment (fiam) [reply testapps/jwks/test/index.test.ts:43]: Thank you!

File: testapps/jwks/test/index.test.ts [hunk 4]
```diff
@@ -88,9 +84,7 @@ describe('test token Authorization', () => {
 
 	test('empty token', async () => {
 		const client = wg.client();
-		client.setExtraHeaders({
-			Authorization: `Bearer `,
-		});
+		client.setAuthorizationToken('');
```
Comment (jensneuse) [inline testapps/jwks/test/index.test.ts:87]: I don't want to annoy you, but wouldn't it make sense to also add client.unsetAuthorizationToken which removes the Authorization header from the extra headers if it existed? setAuthorizationToken with empty string will still send "Authorization: Bearer "

**Review by jensneuse — CHANGES_REQUESTED** [review]

File: testapps/jwks/test/index.test.ts (hunk 4 above)
Comment (fiam) [reply testapps/jwks/test/index.test.ts:87]: No worries, I think it's a good idea. I've added `Client.removeAuthorization()` with the idea of eventually using it for both tokens and cookies. PTAL

**Review by jensneuse — APPROVED** [review]

File: testapps/jwks/test/index.test.ts (hunk 4 above)
Comment (jensneuse) [reply testapps/jwks/test/index.test.ts:87]: LGTM

File: testapps/jwks/test/index.test.ts (hunk 1 above)
Comment (jensneuse) [reply testapps/jwks/test/index.test.ts]: LGTM

**Pull request merged**